File: money/blackjack/dealer.cpp
#include "dealer.h"

void Dealer::reset_hand()
{
  delete m_hand;
  m_hand = new Hand(0);
}

void Dealer::print(bool final_print)
{
  (m_cur_playing)? cout << "*": cout << " ";
  cout << "Dealer:";
  m_hand->dealer_print(m_cur_playing, final_print);
  cout << endl;

  return;
}

int Dealer::hit(Card* new_card)
{
  return m_hand->hit(new_card);
}

File: money/blackjack/python/dealer.py
from hand import *

class Dealer:
  def __init__(self):
    self.m_hand = None

  def new_hand(self):
    del self.m_hand
    self.m_hand = Hand(0)

  def waiting_str(self):
    return " Dealer: " + self.m_hand.dealer_waiting_str()

  def playing_str(self):
    return " Dealer: " + self.m_hand.dealer_playing_str()
